src/shogun/distance/MahalanobisDistance.cpp

```diff
@@ -39,20 +37,43 @@ bool CMahalanobisDistance::init(CFeatures* l, CFeatures* r)
 {
 	CRealDistance::init(l, r);
 
+	REQUIRE(
+	    lhs->get_feature_class() == C_DENSE,
+	    "Left hand side (was %s) has to be CDenseFeatures instance.\n",
+	    lhs->get_name());
+	REQUIRE(
+	    rhs->get_feature_class() == C_DENSE,
+	    "Right hand side (was %s) has to be CDenseFeatures instance.\n",
+	    rhs->get_name());
+
+	REQUIRE(
```

Comment:
and actually this is implicitly tested in `CRealDistance::init`
https://github.com/vinx13/shogun/blob/39e7dd2f390d13e665678be23b0b0221f1212033/src/shogun/distance/RealDistance.h#L34

Reply:
haha I wrote the same thing above ;) sorry didnt see